proto/generate_protos.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 Zenauth Ltd.
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(dirname "$SCRIPT_DIR")"
+PROTO_DIR="${SCRIPT_DIR}/defs"
+CERBOS_MODULE=${CERBOS_MODULE:-"buf.build/cerbos/cerbos-api"}
+
+echo ">> Retrieving proto definitions"
+rm -rf $PROTO_DIR
+(
+    cd $SCRIPT_DIR
+    buf mod update
+    buf export $CERBOS_MODULE -o ${PROTO_DIR}
+)
+
+# `dist` is reserved for housing the python wheel for distribution, so we need to check for namespace collisions
+if [ -d "${PROTO_DIR}/dist" ]; then
+    echo "Error: There cannot be a directory called \`dist\` in ${PROTO_DIR}, as we reserve this directory name for building and distribution. Please talk to the package maintainers. Exiting."
+    exit 1
+fi
+
+echo ">> Generating python classes"
+(
+    cd $ROOT_DIR
+    find $PROTO_DIR -name '*.proto' | xargs -I{} ./pw pdm run python -m grpc_tools.protoc -I $PROTO_DIR --python_out=pyi_out:. --grpc_python_out=. {}
```

Comment:
Just curious: can't we use Buf to directly generate code without downloading the protobufs? 

I have done something like the following before. However, I am not sure whether it has any drawbacks compared to using `grpc_tools.protoc`.

```sh
buf generate --template=buf.gen.yaml https://github.com/cerbos/cerbos.git
```

```yaml
# buf.gen.yaml
version: v1
plugins:
  - plugin: buf.build/protocolbuffers/python
    out: gen/python
  - plugin: buf.build/grpc/python
    out: gen/python
```


https://buf.build/docs/bsr/remote-plugins/usage/#2.-create-a-buf.gen.yaml

Reply:
Ah this is cool, thanks! Originally it was because I didn't see a way to generate the `.pyi` files for type definitions, but that can be added via [this](https://buf.build/protocolbuffers/pyi) 🙌